active/ItemsRepeater/ItemsRepeater-FirstVisibleIndex.md

```diff
@@ -0,0 +1,118 @@
+
+# Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+
+`RealizedItems` and `VisibleItems` are properties that give you access to all realized or visible items within an ItemsRepeater. These two properties both return lists of [UIElement](https://docs.microsoft.com/uwp/api/windows.ui.xaml.uielement?view=winrt-19041) objects that make up the ItemsRepeater. 
+
+Realized items includes all items that are currently visible and all items that are in the cache/buffer area. The size of these buffer areas are configurable via the [HorizontalCacheLength](https://docs.microsoft.com/uwp/api/microsoft.ui.xaml.controls.itemsrepeater.horizontalcachelength?view=winui-2.4) and [VerticalCacheLength](https://docs.microsoft.com/uwp/api/microsoft.ui.xaml.controls.itemsrepeater.verticalcachelength?view=winui-2.4) properties
+
+Visible items refers to all items that are visible in the ItemsRepeater, whether they are fully or partially visible.
+
+# Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+```csharp
+/* Example 1: Check if a specific item is visible by the user. If visible, inform the user that they've scrolled to the bottom of the list. */
+int repeater_length = repeater.ItemsSource.Count;
+
+foreach (Grid item in repeater.VisibleItems){
+  if (repeater.GetElementIndex(item) == repeater_length - 1){
+    myTextBlock.Text = "No more elements to display.";
+  }
+}
+
+/* Example 2: Change the background color of any items that have been seen by the user (i.e. they're visible on the screen). This example applies well to messaging scenarios where the styling is changed for messages that have been read/seen.
+
+By using the RealizedItems property, any item that becomes visible while scrolling will have this changed background color.  */ 
```

Comment:
Great point - not sure why I'm struggling with these samples. See fix in upcoming commit.

Reply:
Those samples are good to go now I would say. Something that we could also have as a sample is how to use the realized items list to update the items when the selection has changed to update the selection indication. Pseudocode:

```
var repeater = new ItemsRepeater(){/* necessary init */};
var selectionModel = new SelectionModel(){ /* necessary init */};
selectionModel.OnSelectionChanged = OnSelectionChanged;

void OnSelectionChanged(){
  foreach(var item in repeater.RealizedItem)
  {
    // Don't know if DataContext is correct here
    if(item.DataContext == selectionModel.SelectedItem)
    {
      item.BorderColor = Colors.Green;
    }
    else
    {
      item.BorderColor = Colors.Transparent;
    }
  }
}
```
However such a sample might be a bit to "complex" or "large" here? What do you think @anawishnoff?

Reply:
Hmm, I'm a bit hesitant about including a sample here that uses SelectionModel just because we don't have public APIs/documentation/an approved spec for it yet, which might lead to it sparking more confusion than clarity. 

@ranjeshj Would you agree?

Reply:
Oh, right, totally forgot that. Including preview APIs in samples is definitely problematic, especially if those might change later.